Sources/PrinceOfVersions/Common/ConfigurationData.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  ConfigurationData.swift
+//  PrinceOfVersions
+//
+//  Created by Ivana Mršić on 16/04/2020.
+//  Copyright © 2020 Infinum Ltd. All rights reserved.
+//
+
+import Foundation
+
+// MARK: - ConfigurationData -
+
+struct ConfigurationData: Decodable {
+    let requiredVersion: Version?
+    let lastVersionAvailable: Version?
+    let notifyLastVersionFrequency: NotificationType?
+    let requirements: Requirements?
+    let meta: [String: AnyDecodable]?
+}
```

Comment:
I would say yes.

It is possible for users to define configuration data that does not apply to the structure here provided because they want to support older versions of PoV (<4.0). Older formatting uses same key in JSON as when we only have new formatting (`ios` or `macos`) so that is the reason I keep all optional.

cc: @jabou

Reply:
I think that some of those values are ok to be non-optional 🤔  
`lastVersionAvailable` - if not found in json, this will fall back to the installed version that is always available
 `notifyLastVersionFrequency` - by default, this is set to `alway`

while `requredVersion`, `requrements` and `meta` should be optional